pkg/pillar/cmd/zedagent/reportinfo.go

```diff
@@ -336,6 +339,58 @@ func PublishDeviceInfoToZedCloud(ctx *zedagentContext) {
 			swInfo)
 	}
 
+	// Reporting all zpools in Strorage Info
+	if vault.ReadPersistType() == types.PersistZFS {
+		zfsVersion, err := zfs.GetZfsVersion()
+		if err != nil {
+			log.Errorf("error: %v", err)
+		}
+
+		zpoolList, err := libzfs.PoolOpenAll()
+		if err != nil {
+			log.Errorf("get zpool list failed %v", err)
+		} else {
+			for _, zpool := range zpoolList {
+				rStorageInfo := new(info.StorageInfo)
+				curentRaid := info.StorageRaidType_STORAGE_RAID_TYPE_NORAID
+				zpoolName := zpool.Properties[libzfs.PoolPropName].Value
+				zpoolSizeInByte, err := strconv.ParseUint(zpool.Properties[libzfs.PoolPropSize].Value, 10, 64)
+				if err != nil {
+					log.Errorf("error with ParseUint for get zpool size in byte: %v", err)
+					zpoolSizeInByte = 0
+				}
+
+				dedup, _ := strconv.ParseFloat(zpool.Properties[libzfs.PoolPropDedupratio].Value, 64)
+				if err != nil {
+					log.Errorf("error with ParseFloat for get compression ratio: %v", err)
+					dedup = 0
+				}
+
+				vdevs, err := zpool.VDevTree()
+				if err != nil {
+					log.Errorf("error with get about RAID info from devTree: %v", err)
+				} else {
+					curentRaid = zfs.GetZpoolRaidType(vdevs)
+				}
+
+				zpool.Close()
+				rStorageInfo.PoolName = *proto.String(zpoolName)
+				rStorageInfo.StorageType = info.StorageTypeInfo_STORAGE_TYPE_INFO_ZFS
+				rStorageInfo.ZpoolSize = *proto.Uint64(zpoolSizeInByte)
+				rStorageInfo.ZfsVersion = *proto.String(zfsVersion)
+				rStorageInfo.CurentRaid = curentRaid
+				rStorageInfo.CompressionRatio = *proto.Float64(dedup)
+				ReportDeviceInfo.StorageInfo = append(ReportDeviceInfo.StorageInfo, rStorageInfo)
+				log.Tracef("report metrics sending info for ZFS zpool %s", zpoolName)
+			}
+		}
+	} else {
+		xStorageInfo := new(info.StorageInfo)
+		xStorageInfo.StorageType = info.StorageTypeInfo_STORAGE_TYPE_INFO_EXT4
```

Comment:
I wonder if we should fill in default values for some of the other fields. For instance, with no compression the compression ratio is 1.0, and there might be a Raid value for no raid at all. Don't know if such defaults will be helpful at the UI.

Reply:
@vk-en would that make sense?

Reply:
I think that we should not mislead anyone with data that we do not collect. In this case, in my opinion, it would be much more correct to process it in the same cloud if any information was not received. For example, output "--"

Well, it's probably obvious that with ext4 there will be no RAID, and there is no compression there either, so I think that we should not send it.